src/Service.Tests/Unittests/SerializationDeserializationTests.cs

```diff
@@ -276,6 +278,120 @@ public void TestDictionaryDatabaseObjectSerializationDeserialization()
+
+            // Create expected JSON
+            string expectedJson = @"
+            {
+                ""stringKey"": ""Hello"",
+                ""numberKey"": 123,
+                ""floatKey"": 123.45,
+                ""booleanKey"": true,
+                ""nullKey"": null,
+                ""arrayKey"": [""Hello"", 123, 123.45, true, null],
+                ""objectKey"": {
+                    ""stringKey"": ""Hello"",
+                    ""numberKey"": 123,
+                    ""floatKey"": 123.45,
+                    ""booleanKey"": true,
+                    ""nullKey"": null
+                }
+            }";
+
+            // Compare the output JSON with the expected JSON
+            Assert.IsTrue(JToken.DeepEquals(JToken.Parse(expectedJson), JToken.Parse(outputJson)));
+        }
+
+        /// <summary>
+        /// This test checks that the ReplaceEnvVarsInJson method throws a JsonException when given an invalid JSON string.
+        /// The test asserts that the exception's message matches the expected error message.
+        /// </summary>
+        [TestMethod]
+        public void TestEnvVariableReplacementWithInvalidJson()
+        {
+            // Create an invalid JSON string
+            string inputJson = @"{ ""stringKey"": ""@env('ENV_STRING')', }";
+
+            // Call the method under test and expect a JsonException
+            try
+            {
+                Utf8JsonReaderExtensions.ReplaceEnvVarsInJson(inputJson);
+                Assert.Fail("Expected a JsonException to be thrown");
+            }
+            catch (JsonException ex)
+            {
+                Assert.AreEqual("Failed to replace environment variables in given JSON. "
+                    + "Expected end of string, but instead reached end of data. LineNumber: 0 | BytePositionInLine: 38.", ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// This test validates that the `ReplaceEnvVarsInJson` method throws a JsonException when
+        /// it encounters a placeholder for an unset environment variable in the provided JSON string.
+        /// The test asserts that the exception's message matches the expected error message.
+        /// </summary>
+        [TestMethod]
+        public void TestEnvVariableReplacementWithMissingEnvVar()
+        {
+            // Create a JSON string with a placeholder for the missing environment variable
+            string inputJson = @"{ ""stringKey"": ""@env('MISSING_ENV_VAR')"" }";
+
+            // Call the method under test and expect a JsonException
+            try
+            {
+                Utf8JsonReaderExtensions.ReplaceEnvVarsInJson(inputJson);
+                Assert.Fail("Expected a JsonException to be thrown");
+            }
+            catch (JsonException ex)
+            {
+                Assert.AreEqual("Failed to replace environment variables in given JSON. "
+                    + "Environment variable 'MISSING_ENV_VAR' is not set.", ex.Message);
```

Comment:
well, actually not. environment files are read as test.  environment values are read as string,i.e. "null", and then added to the json as a null value. So, if a user wants to set a null value. it will be assigned as null and no error should be thorwn. there is a test for the same.